geoutils/raster/raster.py

```diff
@@ -958,6 +1019,68 @@ def to_rio_dataset(self) -> rio.io.DatasetReader:
         # Then open as a DatasetReader
         return mfh.open()
 
+    @classmethod
+    def convert_roi(
+        cls: type[RasterType],
+        roi: dict[str, float | int],
+        transform: tuple[float, ...] | Affine,
+        height: int,
+        width: int,
+        crs: CRS | int,
+    ) -> None:
+        """
+        Convert ROI into the same CRS as the raster and convert the ROI to pixel-based coordinates.
+
+        :param roi: Optional region of interest. Can be pixel-based (dict with keys: 'x', 'y', 'w', 'h')
+            or georeferenced (dict with keys: left', 'bottom', 'right', 'top', optional 'crs').
+        :param transform: Affine 2D transform. Either a tuple(x_res, 0.0, top_left_x,
+            0.0, y_res, top_left_y) or an affine.Affine object.
+        :param height: height in pixels.
+        :param width: width in pixels.
+        :param crs: Coordinate reference system. Any CRS supported by Pyproj (e.g., CRS object, EPSG integer).
+        """
+        if not isinstance(transform, Affine):
+            transform = Affine(*transform)
+
+        roi_pix = {"x", "y", "w", "h"}
+        roi_georef = {"left", "bottom", "right", "top"}
+
+        # Check if ROI is georeferenced or pixel_based and convert to crs
+        if roi_georef.issubset(roi.keys()):
+            crs_roi = roi.get("crs", "EPSG:4326")
+            transformer = Transformer.from_crs(crs_roi, crs, always_xy=True)
```

Comment:
Note that we have a function to convert points from one CRS to another, which saves a few lines: https://github.com/GlacioHack/geoutils/blob/main/geoutils/projtools.py#L246